scrooge-core/src/main/scala/com/twitter/scrooge/TUnboundedByteArrayOutputStream.scala

```diff
@@ -0,0 +1,14 @@
+package com.twitter.scrooge
+
+import org.apache.thrift.TByteArrayOutputStream
+
+/**
+ * A version of ByteArrayOutputStream that exposes the internal buffer.
+ * It's used to replace TByteArrayOutputStream so that the internal buffer won't be reallocated
+ * on reset.
+ */
+private[scrooge] class TUnboundedByteArrayOutputStream(size: Int) extends TByteArrayOutputStream(size) {
+  override def reset() {
```

Comment:
style nitpick: scala is moving away from procedure style methods. can you annotate the return type? `: Unit = {`